apps/meteor/client/views/room/body/hooks/useFileUploadDropTarget.ts

```diff
@@ -55,7 +56,8 @@ export const useFileUploadDropTarget = (): readonly [
 		const uniqueFiles = getUniqueFiles();
 
 		const uploads = Array.from(uniqueFiles).map((file) => {
-			Object.defineProperty(file, 'type', { value: mime.lookup(file.name) });
+			const fileMimeType = mime.lookup(file.name);
+			Object.defineProperty(file, 'type', { value: fileMimeType ?? (rejectUnknownMediaTypes ? undefined : 'application/octet-stream') });
```

Comment:
Any reason for doing this instead of copying the object with the new prop? 🤔

Reply:
> Any reason for doing this instead of copying the object with the new prop? 🤔

I understand using `Object.defineProperty` avoids reassignments to this property, but I can't spot a specific reason for doing this here :thinking: I'll assign the property directly, then
Let's see what front-enders think of this too, I may be missing something

Reply:
> Any reason for doing this instead of copying the object with the new prop? 🤔

I don't think so :thinking: the objects in the files array are not reused by the function that calls `onFileDrop`
So I think it's ok to copy the object here instead of assigning a prop to it

Reply:
I tried returning `return { ...file, type: typeof fileMimeType === 'string' ? fileMimeType : 'application/octet-stream' };` instead of assigning the property but it caused some errors. So I assume that the client depends on the `file` objects that are being changed here
I'd say it has something to do with this file: `apps/meteor/client/lib/chats/flows/uploadFiles.ts`
So I kept this property definition as it was

Reply:
> also, what if lookup returns empty string. ?? is being used here

Fixed! Thanks for the heads-up